src/pudl/extract/excel.py

```diff
@@ -195,6 +196,17 @@ def add_data_maturity(self, df: pd.DataFrame, page, **partition) -> pd.DataFrame
             maturity = "provisional"
         elif self._dataset_name == "eia860m":
             maturity = "monthly_update"
+        elif self._dataset_name == "eia923":
+            # this conditional is a hacky way to get around the fact that the formatting
+            # for the file names didn't always look like "EIA923_Schedules_2_3_4_5_M_"
+            # But, because we only want YTD data, it's fine to set a year floor.
+            if partition["year"] > 2022:
+                release_month = re.search(
+                    r"EIA923_Schedules_2_3_4_5_M_(\d{2})",
+                    self.excel_filename(page, **partition),
+                ).group(1)
```

Comment:
if the filename for the YTD file always contains `EIA923_Schedules_2_3_4_5_M_` you could do something like:
```suggestion
            file_name = self.excel_filename(page, **partition)
            ytd_file_name_start = "EIA923_Schedules_2_3_4_5_M_"
            if ytd_file_name_start in file_name:
                release_month = re.search(
                    r"ytd_file_name_start(\d{2})", file_name,
                ).group(1)
```

i think you could probably even use the `if ytd_file_name_start in file_name:` as the `elif` instead of checking for 923 first.

Reply:
This makes sense, but also it won't fail if there is no capture group found. I feel like it would be good if this failed if, for example, the file naming changed and we needed to come up with a new way to grab it.